packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/catalog/services.py

```diff
@@ -191,3 +196,42 @@ async def _call(
         service_key=service_key,
         service_version=service_version,
     )
+
+
+@log_decorator(_logger, level=logging.DEBUG)
+async def batch_get_my_services(
+    rpc_client: RabbitMQRPCClient,
+    *,
+    product_name: ProductName,
+    user_id: UserID,
+    ids: list[
+        tuple[
+            ServiceKey,
+            ServiceVersion,
+        ]
+    ],
+) -> list[MyServiceGet]:
+    """
+    Raises:
+        ValidationError: on invalid arguments
+        CatalogForbiddenError: no access-rights to list services
+    """
+
+    @validate_call()
+    async def _call(
+        product_name: ProductName,
+        user_id: UserID,
+        ids: list[tuple[ServiceKey, ServiceVersion]],
+    ):
+        return await rpc_client.request(
+            CATALOG_RPC_NAMESPACE,
+            TypeAdapter(RPCMethodName).validate_python("batch_get_my_services"),
+            product_name=product_name,
+            user_id=user_id,
+            ids=ids,
+            timeout_s=40 * RPC_REQUEST_DEFAULT_TIMEOUT_S,
+        )
+
+    result = await _call(product_name=product_name, user_id=user_id, ids=ids)
```

Comment:
Now that I see this pattern I don't prefer it. Since you are not validating on the server side but on the client side.
`@validate_call(config={"arbitrary_types_allowed": True})` can be used server side.